nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/FlowResource.java

```diff
@@ -2084,7 +2086,45 @@ public Response getComponentHistory(
             )
             @PathParam("componentId") final String componentId) {
 
-        authorizeFlow();
+        serviceFacade.authorizeAccess(lookup -> {
+            final NiFiUser user = NiFiUserUtils.getNiFiUser();
+
+            // authorize the flow
+            authorizeFlow();
+
+            boolean authorized = false;
```

Comment:
@mcgilman I think we can get rid of this 'authorized' variable all together here. Instead, we can just do a 'return' where we have 'authorized = true' and then we can avoid ever even needing to check it. Seems like it would make the code a little simpler & easier to follow - what you do think?